packages/workbox-precaching/src/PrecacheController.ts

```diff
@@ -234,135 +174,77 @@ class PrecacheController {
    * Precaches new and updated assets. Call this method from the service worker
    * install event.
    *
+   * Note: this method calls `event.waitUntil()` for you, so you do not need
+   * to call it yourself in your event handlers.
+   *
    * @param {Object} options
    * @param {Event} options.event The install event.
-   * @param {Array<Object>} [options.plugins] Plugins to be used for fetching
-   * and caching during install.
    * @return {Promise<module:workbox-precaching.InstallResult>}
    */
-  async install({event, plugins}: {
-    event: ExtendableEvent;
-    plugins?: WorkboxPlugin[];
-  }) {
-    if (process.env.NODE_ENV !== 'production') {
-      if (plugins) {
-        assert!.isArray(plugins, {
-          moduleName: 'workbox-precaching',
-          className: 'PrecacheController',
-          funcName: 'install',
-          paramName: 'plugins',
+  install(event: ExtendableEvent) {
+    return waitUntil(event, async () => {
+      const installReportPlugin = new PrecacheInstallReportPlugin();
+      this.strategy.plugins.push(installReportPlugin);
```

Comment:
Is there any use case in which someone might call `install()` manually, instead of having it invoked by the event listener?

If so, if they end up calling it twice, this plugin will be added twice.

(Feel free to ignore if you don't think that's a scenario that's likely to happen.)

Reply:
Hmmm, there's definitely a use case for calling `install()` manually (e.g. you want to get the list of updated and not updated URLs). However, I don't think it's an issue to add the plugin more than once, since it's only used to determine the `install()` method's return value.

Calling the install method more than once—in general—isn't great, but that was always possible. In that case the second run would discover that everything is cached and report no updated URLs.

We could try to prevent calling it more than once, but since it's not really destructive (and users should see in the logs that it's happening twice), I'm not sure it's worth the additional code to prevent the possibility.

WDYT?

Reply:
Okay, we can leave this as-is if it's not a big concern.